### DISCUSSION TITLE
Volume control

### DISCUSSION THREAD

**Discussion by MarcHamamji** [andyholmes/valent#586] [Feature Request]
Hello,

When using GSConnect, I have 2 keyboard shortcuts to trigger a script that changes the phone's volume when something is playing on it, using the MPRIS dbus interface.

I know that I can do the same thing with Valent, but a very nice feature would be to use the default volume buttons to change it. From a very high level, Valent could remap / intercept the volume keys and, if something is playing on the phone, adjust its volume, instead of the host's.

Though I'm not sure about the underlying technical challenges, I would really like such a feature, which would make using both devices more seamless.
Thanks for the great work :smile:

**Comment by andyholmes** [comment]
Sorry, I'm not sure how to do that. It seems difficult to intercept keypresses like that, unless Valent is the focused application. Even then I'm not sure how it would decide which device or player to send event to, or whether it should send volume changes to the system or the media player.

Any hints here would be welcome :slightly_smiling_face:

**Reply by MarcHamamji** [reply]
I'm not familiar with all of the Gnome / Gtk low level stuff 😬, so just giving ideas.

About when to update which media player's volume, I imagine checking if the phone is playing any media (by checking if it has a MPRIS player or whatever internal way there is to check if it is playing media). If yes, update the phone's volume, else the host's.

If it's technically not possible to handle the volume keys stuff, or not worth all the hassles which is very possible, I understand. Again, I'm not a Gnome / Gtk dev, so I don't know anything about the internals of this kind of project.

Anyway, thanks for all the work 😅
